### PR TITLE
Added Sonarqube scanner playbook

### PR DESCRIPTION
**Pull request by samirparhi-dev**
The .md file is ready, but the playbook need some changes here and there . I am in process of doing the changes.

@asa1997 : Reviewers can review only steps .md file . Changes will be pushed soon here.

@sand-hya @NeerajK007 @Aditi8007 : FYI

### CODE REVIEW

File: playbooks/besman-sonarqube_scanning-steps.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+curl -u "${TOKEN}": -X POST "${API_ENDPOINT}" \
+  -d "project=${YOUR_PROJECT_KEY}" \
+  -d "name=${YOUR_PROJECT_NAME}" 
+  ```
+5. Performing Sonar Scanning for a project.
+- Make sure you have sonar-Scanner-CLI installed or you can install it by following  URL : https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/scanners/sonarscanner/
+
+- Navigate to your project directory and run the scanner command:
+```
+sonar-scanner \
```
Comment (asa1997) [inline playbooks/besman-sonarqube_scanning-steps.md:53]: @samirparhi-dev There is already a build tool based command for maven and gradle. In that case, will using a sonar scanner work?

Comment (samirparhi-dev) [reply playbooks/besman-sonarqube_scanning-steps.md:53]: yes , Sonar-scanner will work from your terminal , regardless of you build configuration. If you are talking about running ` graddle scan ` , `maven scan` those the a special configuration in the build configuration in `POM.xml` or `gradlefile`.

**Comment by samirparhi-dev** [comment]
The playbook is modified, @asa1997 kindly review and let me know your thoughts

**Comment by samirparhi-dev** [comment]
@asa1997 : the playbook is being now updated with latest code . it assumes user has "VS Code " installed and it opens up there

**Review by asa1997 — COMMENTED** [review]
@samirparhi-dev 
- please raise the pr from main branch
- please update the playbook-metadata.json file as well
-

**Review by asa1997 — CHANGES_REQUESTED** [review]
@samirparhi-dev 
- please correct the naming convention of the file.
- please raise it from main branch
- please update the playbook-metadata.json file as well.

**Comment by samirparhi-dev** [comment]
@asa1997 -

Few-things, I would like to get clarity before I do changes

1. I am really sorry but I can not raise PR from main branch. It is not a best practice, let me know why? Please point me to the release and branching strategy documentation of this project.

2. Let me know the understanding of version name from a specific tool prospective in the playbook naming convention?

If we use the version name let's say for sonar,there will be too
Many sonar playbook and step file in a single directory which will create ambiguity and confusion for developer to spot the bug and tester to find out the right script for testing. 

However If it is decided from design kindly direct me to that documentation .

3. let me know when we talk about version what does that mean ?

- is the the version according to tool LTS
- is the version with prospects of Author

Please provide me the link to documentation of this.

FYI @panickervinod @harimohanr @sumodgeorge 

@sand-hya @Aditi8007 @NeerajK007

File: playbooks/besman-sast-sonarqube-playbook.sh [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+# License: Same as Repository Licence
+# Usage : Beta-Release
+# Date : 12/04/2024
+
+function __besman_init() {
+    __besman_echo_white "initialising"
+    export ASSESSMENT_TOOL_NAME="sonarqube"
+    export ASSESSMENT_TOOL_TYPE="SAST"
+    export ASSESSMENT_TOOL_VERSION="9.9.4"
+    export ASSESSMENT_TOOL_PLAYBOOK="besman-$ASSESSMENT_TOOL_TYPE-$ASSESSMENT_TOOL_NAME-playbook.sh"
```
Comment (asa1997) [inline playbooks/besman-sast-sonarqube-playbook.sh:13]: "besman-$ASSESSMENT_TOOL_TYPE-$ASSESSMENT_TOOL_NAME-playbook.sh" -> "besman-$ASSESSMENT_TOOL_TYPE-0.0.1-playbook.sh" (0.0.1- version of the playbook)

File: playbooks/besman-sast-sonarqube-playbook.sh [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    
+    export steps_file_name="besman-$ASSESSMENT_TOOL_TYPE-$ASSESSMENT_TOOL_NAME-steps.md"
+    export BESMAN_STEPS_FILE_PATH="$BESMAN_PLAYBOOK_DIR/$steps_file_name"
+
+
```
Comment (asa1997) [inline playbooks/besman-sast-sonarqube-playbook.sh:18]: There was a var array here, and a condition to check if any of the required vars are empty. Please add that as well.

Comment (samirparhi-dev) [reply playbooks/besman-sast-sonarqube-playbook.sh:18]: @asa1997 : I am Sorry, But I will not do any changes unless the above query are addressed

File: playbooks/besman-sast-sonarqube-playbook.sh (hunk 2 above)
Comment (samirparhi-dev) [reply playbooks/besman-sast-sonarqube-playbook.sh:13]: @asa1997 : I am Sorry, But I will not do any changes unless the above query are addressed

**Comment by asa1997** [comment]
> @asa1997 -
> 
> Few-things, I would like to get clarity before I do changes
> 
> 1. I am really sorry but I can not raise PR from main branch. It is not a best practice, let me know why? Please point me to the release and branching strategy documentation of this project.
> 2. Let me know the understanding of version name from a specific tool prospective in the playbook naming convention?
> 
> If we use the version name let's say for sonar,there will be too Many sonar playbook and step file in a single directory which will create ambiguity and confusion for developer to spot the bug and tester to find out the right script for testing.
> 
> However If it is decided from design kindly direct me to that documentation .
> 
> 3. let me know when we talk about version what does that mean ?
> 
> * is the the version according to tool LTS
> * is the version with prospects of Author
> 
> Please provide me the link to documentation of this.
> 
> FYI @panickervinod @harimohanr @sumodgeorge
> 
> @sand-hya @Aditi8007 @NeerajK007



@samirparhi-dev When we check in a new playbook or environment we go with 0.0.1 as the initial version of the file. Its a standard that we follow in the project. When a new change is brought in the file, lets say, a new component is added or a bug is fixed, we will increment the version number.

**Comment by samirparhi-dev** [comment]
please check https://github.com/Be-Secure/besecure-playbooks-store/pull/37 instead

**Pull request closed without merging**